_targets/build.project.riscv64-spike

```diff
@@ -70,21 +70,20 @@ b_build_target() {
 
 		make ${MAKEFLAGS} all
 		CFLAGS="$OLDCFLAGS"
-	popd
+	popd >/dev/null
 
 	cp riscv-pk/build-${TARGET}/bbl $PREFIX_BOOT/phoenix-riscv64-spike.bbl
 }
 
 
 b_image_target() {
 	b_log "Creating image from $PREFIX_ROOTFS"
-	size=`find ${PREFIX_ROOTFS} -type f -exec du -k {} \; | awk 'BEGIN { sum=0 }; { sum+=$1; }; END { print sum }'`
-	size=`expr ${size} \* 150 / 100`
+	size=$(find "${PREFIX_ROOTFS}" -type f -exec du -k {} \; | awk 'BEGIN { sum=0 }; { sum+=$1; }; END { print sum }')
```

Comment:
Instead of call `du` on each file, maybe it will be better:
```
size=$(find "${PREFIX_ROOTFS}" -type f -print0 | du -c -k --files0-from - | awk 'END{print $1}')
```
You can do arithmetic in awk:
```
awk 'END{print($1 * 150) / 100}'
```

Reply:
I think leaving it as is is better - the first line computes the real sum of file sizes and the second just estimates the partition size to be 1.5x larger than that.

Maybe it would increase readability to change variables naming to `fileSizeSum` and `partSize`, but it can be done in a separate PR if desirable.

Reply:
What do you think about the use of find and du as proposed.

Reply:
You're right - I didn't dig into the ext2 image creation too much, the main point of this PR is to make all targets compilable, introduce docker-based build and basic CI using Github Actions.

The ext2 creation needs to be probably revised, as `genext2fs` takes size `-b` in blocks, and we should take into account that block size on host filesystem (eg. Your EXT4) might be different than on the target image.

For now I think we don't have the ability to mount ext2 partition on riscv64 targets, it would be better to focus on ia32-generic on which mounting ext2 works (can be tested under qemu).

This is not related to the buildsystem enhancements, should be done as a separate PR.

Reply:
edit: I've checked that default block size in `genext2fs` is 1024, so we should probably get the ceil of file_size / 1024, eg:
```bash
echo $((($(stat -c "%s" file.txt) + 1023) / 1024))
```

But I'm not proficient enough to make the change and currently don't have the means to test it, so I'm leaving it as-is.

I think @Maxez is more knowledgeable in this topic, maybe our comments would help him.